src/TLuaInterpreter.cpp

```diff
@@ -9334,60 +8921,15 @@ int TLuaInterpreter::setMapZoom(lua_State* L)
 // Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#createMapImageLabel
 int TLuaInterpreter::createMapImageLabel(lua_State* L)
 {
-    // N/U:     int args = lua_gettop(L);
-    if (!lua_isnumber(L, 1)) {
-        lua_pushstring(L, "createMapImageLabel: wrong argument type");
-        return lua_error(L);
-    }
-    int area = lua_tointeger(L, 1);
-
-    if (!lua_isstring(L, 2)) {
-        lua_pushstring(L, "createMapImageLabel: wrong argument type");
-        return lua_error(L);
-    }
-    QString text = lua_tostring(L, 2);
-
-    if (!lua_isnumber(L, 3)) {
-        lua_pushstring(L, "createMapImageLabel: wrong argument type");
-        return lua_error(L);
-    }
-    float posx = lua_tonumber(L, 3);
-
-    if (!lua_isnumber(L, 4)) {
-        lua_pushstring(L, "createMapImageLabel: wrong argument type");
-        return lua_error(L);
-    }
-    float posy = lua_tonumber(L, 4);
-
-    if (!lua_isnumber(L, 5)) {
-        lua_pushstring(L, "createMapImageLabel: wrong argument type");
-        return lua_error(L);
-    }
-    float posz = lua_tonumber(L, 5);
-
-    if (!lua_isnumber(L, 6)) {
-        lua_pushstring(L, "createMapImageLabel: wrong argument type");
-        return lua_error(L);
-    }
-    float width = lua_tonumber(L, 6);
-
-    if (!lua_isnumber(L, 7)) {
-        lua_pushstring(L, "createMapImageLabel: wrong argument type");
-        return lua_error(L);
-    }
-    float height = lua_tonumber(L, 7);
-
-    if (!lua_isnumber(L, 8)) {
-        lua_pushstring(L, "createMapImageLabel: wrong argument type");
-        return lua_error(L);
-    }
-    float zoom = lua_tonumber(L, 8);
-
-    if (!lua_isboolean(L, 9)) {
-        lua_pushstring(L, "createMapImageLabel: wrong argument type");
-        return lua_error(L);
-    }
-    bool showOnTop = lua_toboolean(L, 9);
+    int area = getVerifiedInt(L, __func__, 1, "areaID");
+    QString text = getVerifiedString(L, __func__, 2, "text");
+    float posx = getVerifiedFloat(L, __func__, 3, "posX");
+    float posy = getVerifiedFloat(L, __func__, 4, "posY");
+    float posz = getVerifiedFloat(L, __func__, 5, "posZ");
+    float width = getVerifiedFloat(L, __func__, 6, "width");
+    float height = getVerifiedFloat(L, __func__, 7, "height");
+    float zoom = getVerifiedFloat(L, __func__, 8, "zoom");
+    bool showOnTop = getVerifiedBoolean(L, __func__, 9, "zoom");
 
```

Comment:
:thought_balloon: There is now an option to NOT scale the image - haven't tried playing with it to see how it acts in conjunction with the scaling - but it might be desirable to add the later added feature to the Lua API.